### PR TITLE
feature/server-create-user

### LINKED ISSUE #18: 0.2 Registrierung
Als Benutzer kann ich mich beim Start der Applikation registrieren.

_Akzeptanzkriterien_:
Beim Start der Applikation wird der nicht authentifizierte User auf eine Seite weitergeleitet auf der er sich registrieren oder anmelden kann. Nachdem der User das Registrierungsformular abgesendet hat, sind seine Daten in der Datenbank gespeichert und der User ist angemeldet.

- [ ] Dto definieren
- [ ] Endpoint definieren
- [ ] Service implementieren

### LINKED ISSUE #25: 0.3 Login
Ich als Benutzer kann mich am System anmelden.

Ein User der registriert aber nicht angemeldet ist kann sich nach dem Starten der Applikation einlogen. Nach dem erfolgreichen Login wird eine Übersicht der erfassten Lektionen und Aufgaben angezeigt.

- [ ] Dto definieren
- [ ] Endpoint definieren
- [ ] Service implementieren

### PR DESCRIPTION
**Pull request by jmesserli**
This closes #25 and closes #18.

Summary of changes:
- ​User now has an Id because apparently it does not like pre-filled String ids 🤔 
- Updated Auth0 library
- Fixed frontend not calling backend's `api/user/login`

I made sure to:

- [x] Add and / or update tests
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Update any documentation (javadoc, domain model, erd, ...)
- [x] Follow the contribution guidelines

Please review @needToRoll and @M4R1KU / cc: @outcobra/developers.

### CODE REVIEW

File: frontend/src/app/shared/http/HttpInterceptor.ts [hunk 1]
```diff
@@ -1,21 +1,18 @@
...
-import {Config} from "../../config/Config";
 
 @Injectable()
 export class HttpInterceptor {
 
     private apiNames: string[];
     private defaultApiName: string;
 
-    constructor(private http: Http, private notificationsService: NotificationsService, private config: Config) {
+    constructor(private http: Http, private notificationsService: NotificationWrapperService, private config: Config) {
```
Comment (M4R1KU) [inline frontend/src/app/shared/http/HttpInterceptor.ts]: Why did you changed that. It should work now with a change on my branch.

Comment (jmesserli) [inline frontend/src/app/shared/http/HttpInterceptor.ts]: Well I don't have your branch yet in here. But if you want I can remove it.

Comment (M4R1KU) [inline frontend/src/app/shared/http/HttpInterceptor.ts]: If you undo it, we don't have to change it later😜

**Comment by jmesserli** [comment]
This depends on #41

**Comment by jmesserli** [comment]
Please merge this after #43

File: frontend/src/assets/config/prod.json [hunk 2]
```diff
@@ -1,6 +1,21 @@
 {
     "api": {
-        "apiBase": "https://svmeal-api.jmnw.me/",
-        "context": "api"
+        "defaultApiName": "outcobra",
+        "apis": [
+            {
+                "name": "outcobra",
+                "apiBase": "{TODO}",
```
Comment (needToRoll) [inline frontend/src/assets/config/prod.json]: whats up with the TODOs?

File: frontend/src/assets/config/prod.json [hunk 3]
```diff
@@ -1,6 +1,21 @@
...
+                "authToken": true
+            }
+        ]
+    },
+    "auth0": {
+        "clientID": "cVzf1vE5BOFKOqW8ic8sac0UqM2CqDN4",
+        "domain": "outcobra.eu.auth0.com",
+        "callbackURL": "{TODO}"
```
Comment (needToRoll) [inline frontend/src/assets/config/prod.json]: whats up with the TODOs?

File: frontend/src/assets/config/prod.json (hunk 2 above)
Comment (jmesserli) [inline frontend/src/assets/config/prod.json]: We don't know the productive deploy domain yet.

**Pull request merged**